multiview/embed/splitae.py

```diff
@@ -0,0 +1,241 @@
+                                                self.hiddenSize,
+                                                self.numHiddenLayers,
+                                                view1.shape[1]).to(device)
+        self.view2Decoder_ = _FullyConnectedNet(self.embedSize,
+                                                self.hiddenSize,
+                                                self.numHiddenLayers,
+                                                view2.shape[1]).to(device)
+
+        if self.printGraph:
+            print("Parameter counts: \nview1Encoder: {:,}\nview1Decoder: {:,}"
+                  "\nview2Decoder: {:,}"
+                  .format(self.view1Encoder_.paramCount(),
+                          self.view1Decoder_.paramCount(),
+                          self.view2Decoder_.paramCount())
+                  )
+
+        parameters = [self.view1Encoder_.parameters(),
+                      self.view1Decoder_.parameters(),
+                      self.view2Decoder_.parameters()]
+        optim = torch.optim.Adam(itertools.chain(*parameters),
+                                 lr=self.learningRate)
+        nSamples = view1.shape[0]
+        epochTrainErrors = []
+        epochTestErrors = []
+
+        for epoch in tqdm.tqdm(range(self.trainingEpochs),
+                               disable=(not self.printInfo)):
+            batchErrors = []
+            for batchNum in range(nSamples // self.batchSize):
+                optim.zero_grad()
+                view1Batch = view1[batchNum*self.batchSize:
+                                   (batchNum+1)*self.batchSize]
+                view2Batch = view2[batchNum*self.batchSize:
+                                   (batchNum+1)*self.batchSize]
+                embedding = self.view1Encoder_(view1Batch.to(device))
+                view1Reconstruction = self.view1Decoder_(embedding)
+                view2Reconstruction = self.view2Decoder_(embedding)
+                view1Error = torch.nn.MSELoss()(view1Reconstruction,
+                                                view1Batch.to(device))
+                view2Error = torch.nn.MSELoss()(view2Reconstruction,
+                                                view2Batch.to(device))
+                totalError = view1Error + view2Error
+                totalError.backward()
+                optim.step()
+                batchErrors.append(totalError.item())
+            if self.printInfo:
+                print("Average train error during epoch {} was {}"
+                      .format(epoch, np.mean(batchErrors)))
+            epochTrainErrors.append(np.mean(batchErrors))
+            if validationXs is not None:
+                testError = self._testError(validationXs)
+                if self.printInfo:
+                    print("Average test  error during epoch {} was {}\n"
+                          .format(epoch, testError))
+                epochTestErrors.append(testError)
+
+        if self.printGraph:
```

Comment:
The code here should probably be moved into a separate function for validation testing. Not too important right now though

Reply:
Pytest will pop up a graph so no can do. We can make a test to compare the pixels of the produced graph to a stored reference graph lol

Reply:
srry should have been more clear. Meant that all the print infos should be pulled out as separate helper functions users can call specifically for printing results. Not important right now just thought it'd be better modularized that way.

Reply:
going to go ahead and resolve this for now